### ISSUE TITLE
EP not working for ajax-powered search

### ISSUE THREAD

**Issue by psorensen** [10up/ElasticPress#216]
Perhaps the burden doesn't fall on the plugin development and I'm just doing it wrong.

I'm trying to build an ajax-powered search by sending a request to a function that returns a query. I've made it work no problem with EP disabled, but re-enabling EP seems to have no affect. I'll post the relevant parts of my code for reference as to provide my method used.

***ajax-search.js***

```
// Get Search Form Values
function getSearchValue()
{
    var searchValue = $('#rfp_search').val(); //Get search form text input value
    return searchValue;
}

function run_search(paged){
    var ajax_url = ajax_search_params.ajax_url;

    $.ajax({
        type: 'GET',
        url: ajax_url,
        data: {
            action: 'get_search',
            search: getSearchValue()
        },
        success: function(data)
        {
            $('#search-results').html(data);
        },
        error: function()
        {
            $("#search-results").html('<p>There has been an error</p>');
        }
    });
}
```

***search-functions.php***
```
wp_register_script( 'ajax-search-js', get_stylesheet_directory_uri() . '/library/build/js/ajax-search.js', array( 'jquery' ), '', true );
wp_localize_script( 'ajax-search-js', 'ajax_search_params', array( 'ajax_url' => admin_url( 'admin-ajax.php' ) ) );
wp_enqueue_script( 'ajax-search-js' );

add_action ('wp_ajax_get_search', 'om_ajax_search');
add_action ('wp_ajax_nopriv_get_search', 'om_ajax_search');

function om_ajax_search(){

    $query_data = $_GET;
    $search_value = ($query_data['search']) ? $query_data['search'] : false;

    var_dump(is_search());
    var_dump($tax_query);
    $args = array(
        's' => $search_value,
        'post_type' => 'rfp'
    );
    $loop = new WP_Query($args);

    if( $loop->have_posts() ): ?> 
    <p><?php global $wp_query; echo $wp_query->found_posts.' results found.'; ?></p>
    <?php 
    while( $loop->have_posts() ): $loop->the_post(); ?>
        <h5><a href="<?php the_permalink(); ?>"><?php the_title(); ?></a></h5>
    <?php endwhile;
    else:
        echo '<h2>No Posts Found<h2>';
    endif;
    wp_reset_postdata();
    die();
}
```

... and obviously an empty div on `archive-rfp` with an ID of `search-results` to accept the HTML from the ajax callback.

Like I said, the ajax functionality works great, EP just doesn't seem to affect it in any way. I'm sure this isn't a bug, but maybe someone has an idea of how I could alter this so EP registers it as a search.

**Comment by tlovett1** [comment]
This looks like it should work fine. I'm going to test locally.

**Comment by tlovett1** [comment]
The problem is here. `admin-ajax.php` counts as an admin request. I am going to add a filter so you can enable query integration only for ajax requests. However, you shouldn't make requests to `admin-ajax.php` from the front end since page caching can't be taken advantage of. Instead you should create an endpoint on the front end to hit for search.

**Comment by tlovett1** [comment]
Meant to paste this: https://github.com/10up/ElasticPress/blob/develop/classes/class-ep-wp-query-integration.php#L27

**Comment by psorensen** [comment]
Thanks for the quick fix, @tlovett1. I will heed your advice regarding making requests to `admin-ajax.php`.
